src/schema/artwork/__tests__/artwork.test.js

```diff
@@ -1177,54 +1177,61 @@ describe("Artwork type", () => {
       {
         artwork(id: "richard-prince-untitled-portrait") {
-          domesticShipping
+          shippingInfo
         }
       }
     `
 
-    it("is null when its domestic_shipping_fee_cents is null", () => {
+    it("is set to prompt string when its domestic_shipping_fee_cents is null and international_shipping_fee_cents is null", () => {
       artwork.domestic_shipping_fee_cents = null
+      artwork.international_shipping_fee_cents = null
       return runQuery(query, rootValue).then(data => {
         expect(data).toEqual({
-          artwork: { domesticShipping: null },
+          artwork: {
+            shippingInfo: "Shipping, tax, and service quoted by seller",
+          },
         })
       })
     })
 
-    it("is formatted domestic_shipping_fee_cents when its domestic_shipping_fee_cents is set", () => {
-      artwork.domestic_shipping_fee_cents = 1000
+    it("is set to free shipping string when its domestic_shipping_fee_cents is 0 and international_shipping_fee_cents is 0", () => {
+      artwork.domestic_shipping_fee_cents = 0
+      artwork.international_shipping_fee_cents = 0
       return runQuery(query, rootValue).then(data => {
         expect(data).toEqual({
-          artwork: { domesticShipping: "$10" },
+          artwork: {
+            shippingInfo: "Free shipping worldwide",
+          },
         })
       })
     })
-  })
-
-  describe("#internationalShipping", () => {
-    const query = `
-      {
-        artwork(id: "richard-prince-untitled-portrait") {
-          internationalShipping
-        }
-      }
-    `
 
-    it("is null when its international_shipping_fee_cents is null", () => {
-      artwork.domestic_shipping_fee_cents = null
+    it("is set to domestic shipping only when its domestic_shipping_fee_cents is present and international_shipping_fee_cents is not", () => {
+      artwork.domestic_shipping_fee_cents = 1000
+      artwork.international_shipping_fee_cents = 0
       return runQuery(query, rootValue).then(data => {
-        expect(data).toEqual({ artwork: { internationalShipping: null } })
+        expect(data).toEqual({
+          artwork: {
+            shippingInfo: "Shipping: $10.00 Continental US only",
+          },
+        })
       })
     })
 
-    it("is formatted international_shipping_fee_cents when its international_shipping_fee_cents is set", () => {
-      artwork.international_shipping_fee_cents = 1100
+    it("is set to domestic and intermational shipping when both domestic_shipping_fee_cents and present and international_shipping_fee_cents are set", () => {
+      artwork.domestic_shipping_fee_cents = 1000
+      artwork.international_shipping_fee_cents = 2000
       return runQuery(query, rootValue).then(data => {
-        expect(data).toEqual({ artwork: { internationalShipping: "$11" } })
+        expect(data).toEqual({
+          artwork: {
+            shippingInfo:
+              "Shipping: $10.00 Continental US, $20.00 rest of the world",
```

Comment:
We probably need to explicitly pass in the defaults, calling the resolver like this might be doing something odd with the defaults /shrug

can just be `.resolve(artwork, { precision: 0 }` or something?

Reply:
You can also extract the code in `amount`, into a helper, or even use the [auction results one](https://github.com/artsy/metaphysics/blob/5b25ec8271952f4b8eb7d0f72534007e110e8c5d/src/schema/auction_result.js#L144), in a helper, and avoid the sort-of odd invoking this resolver in an intermediate way.